eidas_node/saml.py

```diff
@@ -543,7 +544,10 @@ def _parse_assertion(self, response: LightResponse, assertion: Element) -> None:
                 try:
                     response.level_of_assurance = LevelOfAssurance(authn_class_elm.text)
                 except ValueError:
-                    raise ValidationError({get_element_path(authn_class_elm): authn_class_elm.text}) from None
+                    try:
+                        response.level_of_assurance = (auth_class_map or {})[authn_class_elm.text]
+                    except KeyError:
+                        raise ValidationError({get_element_path(authn_class_elm): authn_class_elm.text}) from None
```

Comment:
I'd prefer something more legible, along the lines of:

```py
if auth_class_map and auth_class_map.get(authn_class_elm.text):
    response.level_of_assurance = auth_class_map.get(authn_class_elm.text)
else:
    raise ValidationError({get_element_path(authn_class_elm): authn_class_elm.text}) from None
```

Reply:
It should be more legible now.